internal/utils/utils.go

```diff
@@ -14,3 +15,14 @@ func IsTruthy(val string) bool {
 		return false
 	}
 }
+
+// GetIntOrDefFromMap returns the value of a key in a map as an int,
+// or a default value if the key is not present or a valid int.
+func GetIntOrDefFromMap(m map[string]string, key string, def int) int {
```

Comment:
Also, in the other PR (https://github.dev/shubham1172/components-contrib/blob/68704c23e04333e6790ba0924d1c24f83cd1113c/internal/utils/utils.go#L21-L22) there's an `GetUint64OrDefFromMap`... Can the two be implemented just once, with a generic?